Lib/test/test_shutil.py

```diff
@@ -1249,6 +1249,15 @@ def test_copyfile_same_file(self):
         # Make sure file is not corrupted.
         self.assertEqual(read_file(src_file), 'foo')
 
+    @unittest.skipIf(MACOS or _winapi, 'On MACOS and Windows the errors are not confusing (though different)')
```

Comment:
this raises the question of... what are the errors on macOS and Windows?  Shouldn't we make the FileNotFoundError exception class in this situation consistent?

Reply:
The error on MacOS I get is this:

```
python3 ~/temp/a.py
Traceback (most recent call last):
  File "/Users/ak/temp/a.py", line 29, in <module>
    copy('LICENSE', 'a_py/')
  File "/usr/local/Cellar/python@3.9/3.9.1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/shutil.py", line 418, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/local/Cellar/python@3.9/3.9.1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: 'a_py/'
```

So I made it consistent with MacOS, which should make it easier for handling (and understanding) this error for programs that run on both Linux and MacOS.

On Windows, it's a different error (quoting @eryksun):

> Note that in Windows this case fails with EINVAL, which is at least less misleading than EISDIR. The EINVAL error is based on WinAPI ERROR_INVALID_NAME (123).

From here: https://bugs.python.org/issue43219#msg387134

My thinking was to leave Windows alone to keep backwards compatibility and also because the error is not confusing (at least it doesn't seem confusing to me, compared to ISADirError), and generally I think it's good to let Windows raise its own errors because windows developers will be familiar with them. Also if we decide to make Windows error consistent with Linux/Mac, it would make sense to make all OSes' errors also consistent, and that would be a much bigger backwards-incompatible change.